core/src/main/scala/caliban/HttpRequestMethod.scala

```diff
@@ -1,14 +1,23 @@
 package caliban
 
 import caliban.CalibanError.ValidationError
-import zio.{ FiberRef, Unsafe }
+import zio.{ FiberRef, Scope, UIO, URIO, Unsafe, ZIO }
 
 private[caliban] sealed trait HttpRequestMethod
 
 private[caliban] object HttpRequestMethod {
   case object GET  extends HttpRequestMethod
   case object POST extends HttpRequestMethod
 
-  val fiberRef: FiberRef[HttpRequestMethod] = Unsafe.unsafe(implicit u => FiberRef.unsafe.make(POST))
   val MutationOverGetError: ValidationError = ValidationError("Mutations are not allowed for GET requests", "")
+
+  private val fiberRef: FiberRef[HttpRequestMethod] = Unsafe.unsafe(implicit u => FiberRef.unsafe.make(POST))
+
+  val get: UIO[HttpRequestMethod] = fiberRef.get
+
+  def setWith[R, E, B](method: HttpRequestMethod)(zio: ZIO[R, E, B]): ZIO[R, E, B] =
+    method match {
+      case POST => zio
+      case GET  => fiberRef.locally(method)(zio)
+    }
```

Comment:
I changed this a little bit to hide the implementation detail of using a `FiberRef` and no-op for `POST` requests.

For `setWith`, it could also be implemented as below, but _I think_ `locally` would be more performant since it doesn't require adding `Scope` to the environment, and we can no-op it for `POST` requests

```scala
def set(method: HttpRequestMethod): URIO[Scope, Unit] = fiberRef.locallyScoped(method)
```